### ISSUE TITLE
Transitionable.halt() does not stop a previous duration

### ISSUE THREAD

**Issue by roblav96** [dmvaldman/samsara#14]
# EDIT
**Reverting back to 0.0.4 solves this issue.**
*Trying to use the latest 0.1.3, final project below*

I'm trying to make a simple progress indicator with fn.start() fn.stop() and fn.error()

```javascript
function Prog( context ) {
	this.wWidth = window.innerWidth
	this.halfwWidth = this.wWidth * 0.5
	this.wHeight = window.innerHeight
	this.halfwHeight = this.wHeight * 0.5

	this.index = 501
	this.posX = 0
	this.active = false
	this.err = false
	this.width = this.wWidth * 0.25
	this.height = 5

	this.surf = new Surface( {
		size: [ this.width, this.height ],
		origin: [ 0.5, 0 ],
		properties: {
			background: '#11C1F3'
		}
	} )

	this.x = new Transitionable( 0 )
	this.xTrans = this.x.map( function ( v ) {
		return Transform.translateX( v )
	} )

	this.swap = function () {
		if ( this.active == false || this.err == true ) {
			return
		}

		this.posX = ( this.posX ) ? 0 : this.wWidth
		this.x.set( this.posX, {
			duration: 1000,
			curve: 'easeOutBounce'
		}, function () {
			console.warn( 'end swap' )
			this.swap()
		}.bind( this ) )

	}

	this.done = function () { // THIS DOES NOT STOP ANYTHING
		console.warn( 'done' )
		this.x.halt() // doesnt halt
		this.x.set( 0 ) // doesnt set it to 0
		this.x.reset( 0 ) // doesnt do anything
		this.posX = 0
		this.active = false
		// This only stops when the previous transition from this.swap() has completed
	}

	this.start = function () {
		if ( this.active == true || this.err == true ) {
			return
		}
		console.warn( 'start' )

		this.active = true
		this.swap()
	}

	this.stop = function () {
		if ( this.active == false || this.err == true ) {
			return
		}
		console.warn( 'stop' )

		this.done()
	}

	this.error = function () {
		this.err = true
		// make cool red animation
		this.err = false
	}

	context.add( {
		transform: this.xTrans
	} ).add( this.surf )

}

module.exports = Prog
```

Am I doing something wrong here? I'm trying to follow the documents as close as possible.

**Comment by roblav96** [comment]
If i keep pressing the stop button it jumps to 0 for one frame then continues back where it was in the duration transition.

**Comment by roblav96** [comment]
# Final EDIT

```javascript
var Curves = require( './Curves.js' )
var Samsara = require( 'samsarajs' )
var Surface = Samsara.DOM.Surface
var Transform = Samsara.Core.Transform
var MouseInput = Samsara.Inputs.MouseInput
var TouchInput = Samsara.Inputs.TouchInput
var SequentialLayout = Samsara.Layouts.SequentialLayout
var Transitionable = Samsara.Core.Transitionable
var Accumulator = Samsara.Streams.Accumulator



function Prog( context ) {
	this.wWidth = window.innerWidth
	this.halfwWidth = this.wWidth * 0.5
	this.wHeight = window.innerHeight
	this.halfwHeight = this.wHeight * 0.5

	this.index = 501
	this.posX = 0
	this.active = false
	this.err = false
	this.width = this.wWidth * 0.25
	this.height = 5



	this.x = new Transitionable( this.posX )
	this.opa = new Transitionable( 0 )

	this.xTrans = this.x.map( function ( v ) {
		return Transform.translateX( v )
	} )

	this.surf = new Surface( {
		size: [ this.width, this.height ],
		origin: [ 0.5, 0 ],
		opacity: this.opa,
		attributes: {
			id: 'prog_surf'
		},
		properties: {
			background: '#11C1F3'
		}
	} )

	this.swap = function () {
		if ( this.active == false || this.err == true ) {
			return
		}

		this.posX = ( this.posX ) ? 0 : this.wWidth
		this.x.set( this.posX, {
			duration: 1000,
			curve: 'easeOutBounce'
		}, function () {
			if ( this.active == false ) {
				this.x.reset( 0 )
				this.x.set( 0 )
				return
			}

			this.swap()
		}.bind( this ) )
	}

	this.done = function () {
		if ( this.err == true ) {
			return
		}

		this.active = false
		this.posX = 0

		this.opa.reset( 0 )

		this.surf.setProperties( {
			background: '#11C1F3'
		} )

	}

	this.start = function () {
		if ( this.active == true || this.err == true ) {
			return
		}

		this.active = true
		this.posX = 0

		this.opa.set( 1 )

		this.x.reset( 0 )

		this.swap()
	}

	this.stop = function () {
		if ( this.active == false || this.err == true ) {
			return
		}

		this.surf.setProperties( {
			background: '#33CD5F'
		} )

		this.opa.set( 0, {
			duration: 500
		}, function () {
			if ( this.active == false ) {
				return
			}
			this.done()
		}.bind( this ) )
	}

	this.error = function () {
		if ( this.err == true ) {
			return
		}
		
		this.err = true
		this.active = true
		this.surf.setProperties( {
			background: '#EF473A'
		} )

		this.opa.reset( 1 )
		this.opa.set( 1 )
		this.surf.setSize( [ this.wWidth, this.height ] )

		this.x.reset( this.wWidth * 1.5 )

		this.x.set( this.wWidth * 0.5, {
			duration: 1000,
			curve: Curves.outBounce
		}, function () {
			// this.opa.set( 1 )
			this.opa.set( 0, {
				duration: 1000
			}, function () {
				this.surf.setSize( [ this.width, this.height ] )
				this.err = false
				this.done()
			}.bind( this ) )
		}.bind( this ) )

	}

	context.add( {
		transform: this.xTrans
	} ).add( this.surf )

}

module.exports = Prog
```

I'm starting to get frustrated with this. No matter what I do, halt() reset() set() just don't do what they are supposed to do. During a transition, halt doest do anything. reset only sets the position for 1 engine frame as with set.

I really want to help to let me know what I can do.

**Comment by dmvaldman** [comment]
This may be because of a bug in the `.halt` method that was recently fixed in the source files, but not yet bundled into the distribution files. Could you try testing this code using the AMD modules in the `samsara` directory?

**Comment by roblav96** [comment]
I'm building via browserify. Not too familiar with AMD :/

I'll try building from the latest master right now. Thank you!

**Comment by dmvaldman** [comment]
Here's an updated CommonJS bundle: http://samsarajs.org/dist/2.0.0/samsara.js 

Let me know if that fixes your issue

**Comment by roblav96** [comment]
OK I'll let you know when I get back home. I'm AFK, sorta

**Comment by roblav96** [comment]
Your link to v2 is dead. I think I have it on my desktop though.

**Comment by dmvaldman** [comment]
Oops sorry: [http://samsarajs.org/dist/0.2.0/samsara.js](http://samsarajs.org/dist/0.2.0/samsara.js)

**Comment by dmvaldman** [comment]
I'd like to close this issue as I believe it's resolved in 0.2.0. Let me know if you're still having problems

**Comment by roblav96** [comment]
Well I have not been able to test yet. The newest v2 broke a lot of things so I'm staying on 0.1.3 till I have time to play with it on the weekend.

When I say broke I mean nothing on the screen works, except samsara surfaces. I'm sure it's a minor CSS issue.

**Comment by dmvaldman** [comment]
hmm, okay please let me know! Send a screenshot. Hope I didn't break anything :-/

**Comment by roblav96** [comment]
No no, it's ok! I'm gonna look into it tomorrow

**Comment by dmvaldman** [comment]
Closing as this should be fixed is 0.2
